src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala

```diff
@@ -192,7 +192,13 @@ trait OAuthService extends HttpService with PerRequestCreator with FireCloudDire
       .withPath(Uri.Path./)
       .withFragment(s"$redirectFragment?access_token=$accessToken")
 
-    redirect(Uri(uiRedirect.toString), StatusCodes.TemporaryRedirect)
+    // we require a spray DateTime, not a joda DateTime
+    // expire in one day
+    val expiration = spray.http.DateTime.now + (1000 * 60 * 60 * 24)
```

Comment:
Shortening to an hour is fine. I think you're looking for this code: https://github.com/broadinstitute/firecloud-orchestration/blob/develop/src/main/resources/swagger/auth-page.html

If we only place the cookie upon download, we can make it even shorter. I'll work on that.